app/src/main/java/ch/epfl/favo/map/GpsTracker.java

```diff
@@ -57,23 +55,21 @@ public Location getLocation() throws NoPermissionGrantedException, NoPositionFou
       if (isGPSEnabled) provider = LocationManager.GPS_PROVIDER;
       else if (isNetworkEnabled) provider = LocationManager.NETWORK_PROVIDER;
       if (provider != null) {
-        locationManager.requestLocationUpdates(provider, 10000, 10, this);
-        if (locationManager != null) {
-          location = locationManager.getLastKnownLocation(provider);
-          // Log.d("report", "position found by gps");
-        }
+         locationManager.requestLocationUpdates(provider, 10000, 10, this);
```

Comment:
There are a couple of magic numbers in this class, try to extract them out into constants ;)